src/lighteval/pipeline.py

```diff
@@ -211,44 +211,41 @@ def _init_model(self, model_config, model):
     def _init_tasks_and_requests(self, tasks: str):
         with local_ranks_zero_first() if self.launcher_type == ParallelismManager.NANOTRON else nullcontext():
             logger.info("--- LOADING TASKS ---")
+
+            # The registry contains all the potential tasks
             registry = Registry(
                 custom_tasks=self.pipeline_parameters.custom_tasks_directory,
             )
-            task_names_list, fewshots_dict = taskinfo_selector(tasks, registry)
-            task_dict = registry.get_task_dict(task_names_list)
+
+            # load the tasks fro the configs and their datasets
+            task_configs: list[LightevalTaskConfig] = registry.get_tasks_configs(tasks)
+            self.tasks_dict: dict[str, LightevalTask] = registry.get_tasks_from_configs(task_configs)
+            LightevalTask.load_datasets(self.tasks_dict, self.pipeline_parameters.dataset_loading_processes)
+            self.documents_dict = {
+                task.full_name: task.get_docs(self.pipeline_parameters.max_samples)
+                for _, task in self.tasks_dict.items()
+            }
+
+            self.sampling_docs = collections.defaultdict(list)
+            for _, docs in self.documents_dict.items():
+                for doc in docs:
+                    for sampling in doc.sampling_methods:
```

Comment:
does not differentiate between several sampling methods of the same type, no? 
for example, a log prob + target log prob will simply be redirected to loglikelihood